### PR TITLE
Improved the devtools pages

### PR DESCRIPTION
**Pull request by wise4rmgodroot**
## Title
* Improved the devtools pages

## Description

* I created 5 new sub pages for the RPC NODE Provider page (Alchemy, dRPC, getBlock, RPC API, NowNodes)
* Under Each sub pages i explained what they are, why use them and for some i added a simple call for first time users.
* Also i added a button to redirect users to learn more

## Screenshots/GIFs

* N/A

## Testing

* I manually tested the chsanges locally and also used the `yarn build`command to check for broken links.

## Checklist

- [x] I have read and understood the contributing guidelines.
- [x] I have followed the style guide and formatting guidelines.
- [x] I have added appropriate comments to explain the changes.
- [x] I have tested my changes thoroughly.

## Refs

* N/A

### CODE REVIEW

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+sidebar_position: 1
+title: Alchemy
+sidebar_label: Alchemy
+tags: [rsk, rootstock, resources, tutorials,  Blockchains, JSON-RPC, WebSockets, Alchemy, RPC, API, RPC API]
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md:5]: Include mainnet, testnet in the tags if it is supported. Include this on all the tags in the different tools pages

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Alchemy is a leading blockchain developer platform that simplifies the process of building Web3 applications. It provides robust APIs for interacting with blockchains like Ethereum, Polygon, and more.
+
+## Why Use Alchemy?
+- **Scalable infrastructure:** Handle millions of API calls effortlessly.
+- **Reliable and fast:** Built for high performance and minimal downtime.
+- **Comprehensive features:** From querying blockchain data to managing NFTs, Alchemy has it all.
+
+
+## Supported Blockchains
+Alchemy supports multiple blockchains, including:
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md]:
```suggestion
Alchemy supports multiple blockchains, including Rootstock:
```

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+- Ethereum
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md]:
```suggestion
```

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+- Polygon
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md]:
```suggestion
```

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+- Optimism
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md]:
```suggestion
```

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+- Arbitrum
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md:19]:
```suggestion
```

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+## API Features
+- **JSON-RPC Support:** Compatible with standard Ethereum JSON-RPC methods.
+- **Enhanced APIs:** Custom endpoints for performance insights, NFT data, and more.
+- **WebSockets:** For real-time updates and event subscriptions.
+
+## Core API Methods
+1. Get Account Balance
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md:27]: Put these json methods examples in the accordion component and include the response

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+}
+```
+
+:::note[Info]
+
+ To learn more about how to interact with Rootstock network with the Alchemy RPC Provider including the setup and a step by step guide,  
+ 
+ click the button below:
+ 
+<Button href="https://dev.rootstock.io/developers/rpc-api/rootstock/" align="left">Rootstock getBlock Website</Button>
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md:53]: Kindly check the link referenced here and the link title

File: docs/05-dev-tools/node-rpc/dRPC.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+dRPC is a next-generation RPC platform that simplifies and optimizes communication with blockchain nodes. Built for Web3, dRPC provides low-latency, high-reliability access to blockchain data and operations.  
+
+### Why Choose dRPC?  
+- **High Performance:** Optimized for speed and low latency.  
+- **Decentralized:** Built for scalability and resilience.  
+- **Cross-Blockchain Support:** Interact with multiple blockchain ecosystems.  
+  
+
+## Supported Blockchains
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/dRPC.md:17]: Just keep Rootstock remove the others, see comment in Alchemy section

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+{
+  "jsonrpc": "2.0",
+  "method": "eth_getBalance",
+  "params": ["0xAddress", "latest"],
+  "id": 1
+}
+```
+
+2. Send Transaction
+```
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md:38]:
```suggestion
```json
```

File: docs/05-dev-tools/node-rpc/alchemy.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+- Arbitrum
+
+## API Features
+- **JSON-RPC Support:** Compatible with standard Ethereum JSON-RPC methods.
+- **Enhanced APIs:** Custom endpoints for performance insights, NFT data, and more.
+- **WebSockets:** For real-time updates and event subscriptions.
+
+## Core API Methods
+1. Get Account Balance
+```
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/alchemy.md:28]:
```suggestion
```json
```

File: docs/05-dev-tools/node-rpc/getBlock.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
+---
+sidebar_position: 3
+title: GetBlock
+sidebar_label: getBlock
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/getBlock.md:4]:
```suggestion
sidebar_label: GetBlock
```

File: docs/05-dev-tools/node-rpc/getBlock.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+### 1. Sign Up and Get an access Token  
+- Create an account at [GetBlock.io](https://getblock.io).  
+- Generate an [access token](https://getblock.io/docs/get-started/auth-with-access-token/) from your dashboard.  
+
+### 2. Configure Your Endpoint  
+- Base URL:  
+  ```
+  https://{blockchain}.getblock.io/mainnet/YOUR_API_KEY/
+  ```  
+  Replace `{blockchain}` with the desired blockchain name (e.g., `eth`).
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/getBlock.md:41]:
```suggestion
  Replace `{blockchain}` with the desired blockchain name (e.g., `rsk`).
```

File: docs/05-dev-tools/node-rpc/getBlock.md [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+
+
+## **Make Your First Call**  
+
+1. Use a tool like Postman.  
+2. Use the Ethereum endpoint as an example:
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/getBlock.md:47]:
```suggestion
2. Use the Rootstock RPC endpoint as an example:  
```

File: docs/05-dev-tools/node-rpc/getBlock.md [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+     "id": 1
+   }
+   ```  
+   > The `result` field contains the current block number in hexadecimal format.  
+
+:::note[Info]
+
+ To learn more about the getBlock connection with Rootstock, including code snippets, RPC endpoints,  click the button below:
+ 
+<Button href="https://getblock.io/nodes/rsk/" align="left">Rootstock getBlock Website</Button>
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/getBlock.md:75]:
```suggestion
<Button href="https://getblock.io/nodes/rsk/" align="left">Rootstock GetBlock Website</Button>
```

File: docs/05-dev-tools/node-rpc/nownodes.md [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+NowNodes is a Blockchain-as-a-Service (BaaS) platform offering API access to over 50 blockchain networks. It allows developers to connect to full nodes and explore blockchains with minimal setup.  
+
+## Why Choose NowNodes?  
+- **Wide Blockchain Support:** Access popular blockchains like Bitcoin, Ethereum, Binance Smart Chain, and more.  
+- **Free Tier Available:** Start for free with limited API requests.  
+- **Reliable Infrastructure:** High uptime and fast response times.  
+
+
+## Supported Blockchains
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/nownodes.md:18]: Same thing here

File: docs/05-dev-tools/node-rpc/rpc-api.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
+---
+sidebar_position: 5
+title: RPC API
+sidebar_label: RPC API
+tags: [rsk, rootstock, resources, tutorials,  setup, dApps, protocol, RPC, API, ]
+description: "Remote Procedure Call (RPC) is a protocol that allows a program to execute procedures (functions) on a remote server as if they were local calls."
+---  
+ 
+Remote Procedure Call (RPC) is a protocol that allows a program to execute procedures (functions) on a remote server as if they were local calls. This eliminates the need for developers to handle low-level network communication.
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/rpc-api.md:9]: Is this not supposed to reference the Rootstock RPC API?

File: docs/05-dev-tools/node-rpc/rpc-api.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+title: RPC API
+sidebar_label: RPC API
+tags: [rsk, rootstock, resources, tutorials,  setup, dApps, protocol, RPC, API, ]
+description: "Remote Procedure Call (RPC) is a protocol that allows a program to execute procedures (functions) on a remote server as if they were local calls."
+---  
+ 
+Remote Procedure Call (RPC) is a protocol that allows a program to execute procedures (functions) on a remote server as if they were local calls. This eliminates the need for developers to handle low-level network communication.  
+
+### How Our API Works  
+Our API uses the JSON-RPC protocol, a lightweight, text-based standard for defining procedures and their parameters. This makes it easy to communicate with the API using JSON-formatted requests and receive structured responses.
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/rpc-api.md]:
```suggestion
Rootstock RPC API uses the JSON-RPC protocol, a lightweight, text-based standard for defining procedures and their parameters. This makes it easy to communicate with the API using JSON-formatted requests and receive structured responses.  
```

File: docs/05-dev-tools/node-rpc/rpc-api.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+### Supported Features  
+- **Version:** JSON-RPC 2.0  
+- **Methods:** Flexible method definitions for various tasks.  
+- **Error Handling:** Clear error messages and structured codes.  
+
+:::note[Info]
+
+ To learn more about the Rootstock RPC API, including how to use it, features,  click the button below:
+ 
+<Button href="https://dev.rootstock.io/developers/rpc-api/rootstock/" align="left">Rootstock getBlock Website</Button>
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/rpc-api.md:34]: Wrong link title

File: docs/05-dev-tools/node-rpc/rpc-api.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+
+:::
+
+### Key Data Structures  
+- **Request:**
```
Comment (owans) [inline docs/05-dev-tools/node-rpc/rpc-api.md:39]: Request to get what?

File: docs/05-dev-tools/node-rpc/rpc-api.md (hunk 17 above)
Comment (wise4rmgodroot) [reply docs/05-dev-tools/node-rpc/rpc-api.md:9]: No, it doesnt have any reference to Rootstock RPC API? and also if we decide to add a link the page is not talking about RPC in general which is what this page is talking about.

I will add a link to line number 12

File: docs/05-dev-tools/node-rpc/rpc-api.md (hunk 20 above)
Comment (wise4rmgodroot) [reply docs/05-dev-tools/node-rpc/rpc-api.md:39]: If you check this link https://dev.rootstock.io/developers/rpc-api/rootstock/methods/
You will see all of them have same data structure, which is constant
```
{ "jsonrpc": "2.0", "method": "method_name", "params": {}, "id": 1 }
```

Comment (wise4rmgodroot) [reply docs/05-dev-tools/node-rpc/rpc-api.md:39]: So if someone wants to send a request via our JSON RPC, thats the structure they will use.
Take a look at these examples
```
curl --location 'https://rpc.testnet.rootstock.io/<api-key>' \
--request POST \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--data '{
    "jsonrpc":"2.0",
    "method":"eth_accounts",
    "params":[],
    "id":0
}'
```
And this
```
curl --location 'https://rpc.testnet.rootstock.io/<api-key>' \
--request POST \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--data '{
    "jsonrpc":"2.0",
    "method":"eth_blockNumber",
    "params":[],
    "id":0
}'
```

They but have same structure, which is:
```
'{
    "jsonrpc":"2.0",
    "method":"",
    "params":[],
    "id":0
}'
```

**Review by owans — DISMISSED** [review]
LGTM!

**Review by owans — APPROVED** [review]
LGTM!

**Pull request merged**